src/main/java/jnr/posix/SpawnFileAction.java

```diff
@@ -47,7 +53,29 @@ public Open(String path, int fd, int flags, int mode) {
         }
 
         final boolean act(POSIX posix, Pointer nativeFileActions) {
-            return ((UnixLibC) posix.libc()).posix_spawn_file_actions_addopen(nativeFileActions, fd, path, flags, mode) == 0;
+            /*
+            This logic allocates a direct ByteBuffer to use for the path in order to work around systems that have not
+            patched CVE-2014-4043, in which older glibc versions do not make a defensive copy of the file path passed to
+            posix_spawn_file_actions_addopen. The buffer may be freed by the caller before it can be used in an
+            eventual posix_spawn call.
+
+            See https://bugzilla.redhat.com/show_bug.cgi?id=1983750 for a RHEL version of this issue.
+             */
+
+            // determine encoded byte array length
+            CharsetEncoder encoder = Charset.defaultCharset().newEncoder();
```

Comment:
I'm not really familiar with the NIO path encoding framework, but is this really the right way to encode a path in a native byte string?

Reply:
I'm looking into it now, but this is generally right. The main question for me is whether the default charset is the correct one to use for file paths. I will post again with links to how current JDK does this when it handles paths on its own (it is largely done in native code using JNI utility methods).

Reply:
FWIW the original code would have relied on standard JNI mechanisms for encoding the string, but that results in a non-copied buffer we do not control. The logic here attempts to duplicate that logic from Java using a longer-lived native buffer, so we can be sure it lives until after the `posix_spawn` call completes.

Reply:
I have a series of links that should show this is the appropriate way to encode this string, but I will show that as a top-level comment.